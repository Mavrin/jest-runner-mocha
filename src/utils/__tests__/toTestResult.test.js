const toTestResult = require('../toTestResult');

const start = +new Date('2000/01/01');
const filePath = '/path/to/file';

const jestTest = {
  path: 'path/to/file',
};

const passingTest = {
  duration: 1,
  title: 'This test passes[1]',
};

const passingTest2 = {
  duration: 4,
  title: 'This test also passes[2]',
};

const failingSnapshotTest = {
  duration: 2,
  error: {
    avaAssertionError: true,
    source: {
      isDependency: false,
      isWithinProject: true,
      file: '/path/to/file',
      line: 16,
    },
    stack:
      'AssertionError: Did not match snapshot\n    Test.t [as fn] (/path/to/file.js:16:5)\n    ',
    improperUsage: false,
    message: 'Did not match snapshot',
    name: 'AssertionError',
    statements: [],
    values: [
      {
        label: 'Difference:',
        formatted: "- 'Foo BarBaz'\n+ 'Foo Bar Baz'",
      },
    ],
    assertion: 'snapshot',
  },
  title: 'This test failed with a snapshot',
};

const failingTest = {
  duration: 4,
  error: {
    avaAssertionError: true,
    source: {
      isDependency: false,
      isWithinProject: true,
      file: '/path/to/file.js',
      line: 12,
    },
    stack: 'AssertionError\n    Test.t [as fn] (/path/to/file.js:12:5)\n    ',
    improperUsage: false,
    message: '',
    name: 'AssertionError',
    statements: [],
    values: [
      {
        label: 'Difference:',
        formatted: '- 1\n+ 2',
      },
    ],
    assertion: 'deepEqual',
  },
  title: 'This test failed',
};

it('turns a passing ava tests to Jest test result', () => {
  expect(
    toTestResult({
      meta: { start, jestTest },
      stats: {
        file: filePath,
      },
      tests: [passingTest],
    }),
  ).toMatchSnapshot();
});

it('turns a failing ava tests to Jest test result', () => {
  expect(
    toTestResult({
      meta: { start, jestTest },
      stats: {
        file: filePath,
      },
      tests: [failingTest],
    }),
  ).toMatchSnapshot();
});

it('turns a failing snapshot ava tests to Jest test result', () => {
  expect(
    toTestResult({
      meta: { start, jestTest },
      stats: {
        file: filePath,
      },
      tests: [failingSnapshotTest],
    }),
  ).toMatchSnapshot();
});

it('turns a whole ava tests suite to Jest test result', () => {
  expect(
    toTestResult({
      meta: { start, jestTest },
      stats: {
        file: filePath,
      },
      tests: [passingTest, failingSnapshotTest, passingTest2, failingTest],
    }),
  ).toMatchSnapshot();
});
