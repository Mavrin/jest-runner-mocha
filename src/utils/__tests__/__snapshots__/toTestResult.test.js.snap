// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`turns a failing ava tests to Jest test result 1`] = `
Object {
  "console": null,
  "failureMessage": "  [1m[37mThis test failed[39m[22m
  [90m/path/to/file.js:12[39m

  Difference:

  - 1
  + 2
",
  "numFailingTests": 1,
  "numPassingTests": 0,
  "numPendingTests": 0,
  "perfStats": Object {
    "end": 946713600000.004,
    "start": 946713600000,
  },
  "skipped": false,
  "snapshot": Object {
    "added": 0,
    "fileDeleted": false,
    "matched": 0,
    "unchecked": 0,
    "unmatched": 0,
    "updated": 0,
  },
  "sourceMaps": Object {},
  "testExecError": null,
  "testFilePath": "path/to/file",
  "testResults": Array [
    Object {
      "ancestorTitles": Array [],
      "duration": 0.004,
      "failureMessages": "  [1m[37mThis test failed[39m[22m
  [90m/path/to/file.js:12[39m

  Difference:

  - 1
  + 2
",
      "fullName": "This test failed",
      "numPassingAsserts": 1,
      "status": "failed",
      "title": "This test failed",
    },
  ],
}
`;

exports[`turns a failing snapshot ava tests to Jest test result 1`] = `
Object {
  "console": null,
  "failureMessage": "  [1m[37mThis test failed with a snapshot[39m[22m
  [90m/path/to/file:16[39m

  Did not match snapshot

  Difference:

  - 'Foo BarBaz'
  + 'Foo Bar Baz'
",
  "numFailingTests": 1,
  "numPassingTests": 0,
  "numPendingTests": 0,
  "perfStats": Object {
    "end": 946713600000.002,
    "start": 946713600000,
  },
  "skipped": false,
  "snapshot": Object {
    "added": 0,
    "fileDeleted": false,
    "matched": 0,
    "unchecked": 0,
    "unmatched": 1,
    "updated": 0,
  },
  "sourceMaps": Object {},
  "testExecError": null,
  "testFilePath": "path/to/file",
  "testResults": Array [
    Object {
      "ancestorTitles": Array [],
      "duration": 0.002,
      "failureMessages": "  [1m[37mThis test failed with a snapshot[39m[22m
  [90m/path/to/file:16[39m

  Did not match snapshot

  Difference:

  - 'Foo BarBaz'
  + 'Foo Bar Baz'
",
      "fullName": "This test failed with a snapshot",
      "numPassingAsserts": 1,
      "status": "failed",
      "title": "This test failed with a snapshot",
    },
  ],
}
`;

exports[`turns a passing ava tests to Jest test result 1`] = `
Object {
  "console": null,
  "failureMessage": null,
  "numFailingTests": 0,
  "numPassingTests": 1,
  "numPendingTests": 0,
  "perfStats": Object {
    "end": 946713600000.001,
    "start": 946713600000,
  },
  "skipped": false,
  "snapshot": Object {
    "added": 0,
    "fileDeleted": false,
    "matched": 0,
    "unchecked": 0,
    "unmatched": 0,
    "updated": 0,
  },
  "sourceMaps": Object {},
  "testExecError": null,
  "testFilePath": "path/to/file",
  "testResults": Array [
    Object {
      "ancestorTitles": Array [],
      "duration": 0.001,
      "failureMessages": null,
      "fullName": "This test passes[1]",
      "numPassingAsserts": 0,
      "status": "passed",
      "title": "This test passes[1]",
    },
  ],
}
`;

exports[`turns a whole ava tests suite to Jest test result 1`] = `
Object {
  "console": null,
  "failureMessage": "  [1m[37mThis test failed with a snapshot[39m[22m
  [90m/path/to/file:16[39m

  Did not match snapshot

  Difference:

  - 'Foo BarBaz'
  + 'Foo Bar Baz'
",
  "numFailingTests": 2,
  "numPassingTests": 2,
  "numPendingTests": 0,
  "perfStats": Object {
    "end": 946713600000.011,
    "start": 946713600000,
  },
  "skipped": false,
  "snapshot": Object {
    "added": 0,
    "fileDeleted": false,
    "matched": 0,
    "unchecked": 0,
    "unmatched": 1,
    "updated": 0,
  },
  "sourceMaps": Object {},
  "testExecError": null,
  "testFilePath": "path/to/file",
  "testResults": Array [
    Object {
      "ancestorTitles": Array [],
      "duration": 0.001,
      "failureMessages": null,
      "fullName": "This test passes[1]",
      "numPassingAsserts": 0,
      "status": "passed",
      "title": "This test passes[1]",
    },
    Object {
      "ancestorTitles": Array [],
      "duration": 0.002,
      "failureMessages": "  [1m[37mThis test failed with a snapshot[39m[22m
  [90m/path/to/file:16[39m

  Did not match snapshot

  Difference:

  - 'Foo BarBaz'
  + 'Foo Bar Baz'
",
      "fullName": "This test failed with a snapshot",
      "numPassingAsserts": 1,
      "status": "failed",
      "title": "This test failed with a snapshot",
    },
    Object {
      "ancestorTitles": Array [],
      "duration": 0.004,
      "failureMessages": null,
      "fullName": "This test also passes[2]",
      "numPassingAsserts": 0,
      "status": "passed",
      "title": "This test also passes[2]",
    },
    Object {
      "ancestorTitles": Array [],
      "duration": 0.004,
      "failureMessages": "  [1m[37mThis test failed[39m[22m
  [90m/path/to/file.js:12[39m

  Difference:

  - 1
  + 2
",
      "fullName": "This test failed",
      "numPassingAsserts": 1,
      "status": "failed",
      "title": "This test failed",
    },
  ],
}
`;
